cmd/rpcdaemon/commands/trace_filtering.go

```diff
@@ -23,19 +23,73 @@ import (
 )
 
 // Transaction implements trace_transaction
-// TODO(tjayrush): I think this should return an []interface{}, so we can return both Parity and Geth traces
 func (api *TraceAPIImpl) Transaction(ctx context.Context, txHash common.Hash) (ParityTraces, error) {
 	tx, err := api.kv.BeginRo(ctx)
 	if err != nil {
 		return nil, err
 	}
 	defer tx.Rollback()
 
-	traces, err := api.getTransactionTraces(tx, ctx, txHash)
+	txn, _, blockNumber, _ := rawdb.ReadTransaction(ethdb.NewRoTxDb(tx), txHash)
+	if txn == nil {
+		return nil, nil // not error, see https://github.com/ledgerwatch/turbo-geth/issues/1645
+	}
+
+	bn := hexutil.Uint64(blockNumber)
+
+	// Extract transactions from block
+	hash, hashErr := rawdb.ReadCanonicalHash(tx, blockNumber)
+	if hashErr != nil {
+		return nil, hashErr
+	}
+	block, senders, sendersErr := rawdb.ReadBlockWithSenders(ethdb.NewRoTxDb(tx), hash, uint64(bn))
+	if sendersErr != nil {
+		return nil, sendersErr
+	}
+	if block == nil {
+		return nil, nil
+	}
+
+	blockTxs := block.Transactions()
+	if len(blockTxs) != len(senders) {
+		return nil, errors.New("block txs len != senders len")
+	}
+
+	txs := make([]TransactionWithSender, 0, len(senders))
+	for n, tx := range blockTxs {
+		if (tx.Hash() == txHash) {
+			txs = append(txs, TransactionWithSender{
+				tx:     tx,
+				sender: senders[n],
+			})
+		}
+	}
+
+	baseBn := bn
+	if baseBn > 0 {
+		baseBn -= 1
+	}
+
+	traces, err := api.callManyTransactions(ctx, tx, txs, hash, rpc.BlockNumber(baseBn))
```

Comment:
This only needs to execute transactions `0..txIndex` and then we only need to take the latest trace

Reply:
Done